nixos/modules/services/security/cfssl.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cfssl;
+in {
+  options.services.cfssl = {
+    enable = mkEnableOption ''
+      Whether to enable the CFSSL CA api-server.
+    '';
+
+    dataDir = mkOption {
+      default = "/var/lib/cfssl";
+      type = types.str;
+      description = "cfssl work directory";
+    };
+
+    address = mkOption {
+      default = "127.0.0.1";
+      type = types.str;
+      description = "Address to bind";
+    };
+
+    port = mkOption {
+      default = 8888;
+      type = types.int;
+      description = "Port to bind";
+    };
+
+    ca = mkOption {
+      default = "file:${cfg.dataDir}/ca.pem";
```

Comment:
I think doing it like this would make the module system rebuild the man pages and stuff when somebody changes `cfg.dataDir`. Using `defaultText = "file:\${cfg.dataDir}/ca.pem", removing the `default = `, and setting `services.cfssl.ca = "file:${cfg.dataDir}/ca.pem"` in the config section at the bottom should fix that.

Same with the `caKey` option.